api/src/main/java/org/openmrs/module/appframework/context/ProgramConfiguration.java

```diff
@@ -0,0 +1,415 @@
+package org.openmrs.module.appframework.context;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.codehaus.jackson.annotate.JsonProperty;
+import org.openmrs.Concept;
+import org.openmrs.Program;
+import org.openmrs.ProgramWorkflow;
+import org.openmrs.ProgramWorkflowState;
+import org.openmrs.api.APIException;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.appframework.AppFrameworkUtil;
+
+public class ProgramConfiguration {
+	
+	@JsonProperty
+    private String programRef;
+	
+	@JsonProperty
+    private String workflowRef;
+	
+	@JsonProperty
+    private String stateRef;
+	
+	private ResolvedConfiguration resolvedConfig;
+		
+	private List<ProgramWorkflow> allPossibleWorkflows;
+	
+	private List<ProgramWorkflowState> allPossibleStates;
+	
+    public ProgramConfiguration() {
+    	
+    }
```

Comment:
Yes, we need this constructor since we have another constructor defined.

It looks like:

```java
public ResolvedConfiguration(Program program, ProgramWorkflow workflow, ProgramWorkflowState state) {
			this.program = program;
			this.workflow = workflow;
			this.state = state;
}
```